### PR TITLE
Brings `main` branch back to Tendermint 0.34

### PR DESCRIPTION
**Pull request by hansl**
Prod should not use `main` after this PR is merged in. I don't think prod needs to keep up too much though, so we could either pause upgrades until tm34 is officially supported by all nodes, or branch off `main` before this and cherry-pick which fixes we need in prod.

### CODE REVIEW

File: scripts/run.sh [hunk 1]
```diff
@@ -9,7 +9,7 @@
 # $ cd /path/to/many-ledger
 # $ ./script/run.sh
 set -pube # .. snicker ..
-TM_MIN_VERSION=0.35.0 # The minimum tendermint version supported
+TM_MIN_VERSION=0.34.0 # The minimum tendermint version supported
```
Comment (fmorency) [inline scripts/run.sh:12]: nit: we should add a check that errors if we use TM 0.35

File: docker/e2e/update_config.sh [hunk 2]
```diff
@@ -94,17 +96,18 @@ for node in $(seq 0 "$NB_NODES"); do
 
   peer_ids=$(seq 0 "$NB_NODES" | grep -v "$node")
   peers=$(for peer in $peer_ids; do
-    node_id=$(jq -r .id < "${config_root//%/$peer}"/node_key.json)
+    node_id=$(docker run -u $(id -u):$(id -g) -i --rm -v "${tm_root//%/$peer}:/tendermint" tendermint/tendermint:v0.34.24 show_node_id | tail -n +2 | tr -d '\n')
```
Comment (fmorency) [inline docker/e2e/update_config.sh:99]: nit: @fmorency to fix this in his Bazel refactor

**Review by fmorency — APPROVED** [review]
Small nit. I don't really care for it.

LGTM! Thanks!

File: scripts/run.sh (hunk 1 above)
Comment (hansl) [reply scripts/run.sh:12]: This is not overridable by external env variables, so...

Comment (hansl) [reply scripts/run.sh:12]: Done.

**Review by fmorency — APPROVED** [review]

**Pull request merged**